### PR TITLE
polaris-icons v4

### PR DESCRIPTION
**Pull request by alex-page**
### WHY are these changes introduced?

Updates polaris-react to the lastest version of `polaris-icons` to use filled icons. This is a part of https://github.com/Shopify/polaris-ux/issues/432

### WHAT is this pull request doing?

- [x] Update `polaris-icons` to version 4.0.0
- [x] Removes `images.tsx`
- [x] Replaces `filled`, `twotone` and `duotone` major variants to major icons for all components
- [x] Replaces `Dropzone` example illustration with filled icon

### How to 🎩

Look at the visual regression test check and go through affected components.

### 🎩 checklist

* [x] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
* [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
* [x] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
* [x] Updated the component's `README.md` with documentation changes
* [x] [Tophatted documentation](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting%20documentation.md) changes in the style guide
* [x] For visual design changes, pinged one of @ HYPD, @ mirualves, @ sarahill, or @ ry5n to update the Polaris UI kit

### CODE REVIEW

File: src/components/Banner/Banner.tsx [hunk 1]
```diff
@@ -211,45 +211,39 @@ function useBannerAttributes(
...
         ariaRoleType: 'status',
       };
 
     case 'info':
       return {
         defaultIcon: newDesignLanguage
-          ? CircleInformationMajorFilled
-          : CircleInformationMajorTwotone,
+          ? CircleInformationMajor
+          : CircleInformationMajor,
```
Comment (kyledurand) [inline src/components/Banner/Banner.tsx:223]: Looks like we don't need these ternaries any more

**Review by kyledurand — APPROVED** [review]
Code looks good just remove those ternaries and 🚢

**Comment by kaelig** [comment]
You'll want to release an RC2 as some updates have been made to icons after RC1 was released.

Would this PR also be an appropriate place to remove the concept of secondary color in the recolor-icon mixin (I believe it mostly had to do with twotone icons)?

**Review by BPScott — COMMENTED** [review]
All looking good to me.

I'll +1 Kaelig's "we should do a quick RC2 of icons" as there was some build tweaks after v1. Shouldn't have any effect but always nice to confirm. Do we want to ship a v4 final before merging this or are you thinking 

As a heads up: when updating this in styleguide you'll need to update the icon names in some places there too:

- For codesandbox editing: https://github.com/Shopify/polaris-styleguide/blob/cabfddfebacc3c4fb103cf80462023d5261c8ac0/app/components/Playground/components/Sandbox/Sandbox.tsx#L13
- For the in-page previews: https://github.com/Shopify/polaris-styleguide/blob/cabfddfebacc3c4fb103cf80462023d5261c8ac0/client/demo-app/Demo/Demo.tsx#L11

**Comment by alex-page** [comment]
@BPScott strategy for the release is here https://github.com/Shopify/polaris-ux/issues/432

**Review by kyledurand — APPROVED** [review]
Looks good to me 👍

**Review by BPScott — APPROVED** [review]
Yay!

**Pull request merged**